src/app/routers/schema_router.py

```diff
@@ -74,9 +77,51 @@ async def get_schema_from_bucket(
         logger.error("Schema metadata not found")
         raise exceptions.ExceptionNoSchemaFound
 
-    logger.info("Bucket schema metadata location successfully retrieved.")
-    logger.debug(f"Bucket schema metadata location: {bucket_schema_filename}")
-    logger.info("Getting schema metadata...")
+    logger.info("Bucket schema location successfully retrieved.")
+    logger.debug(f"Bucket schema location: {bucket_schema_filename}")
+    logger.info("Getting schema...")
+
+    schema = schema_bucket_repository.get_schema_file_as_json(bucket_schema_filename)
+
+    logger.info("Schema successfully retrieved.")
+    logger.debug(f"Schema: {schema}")
+
+    return schema
+
+
+@router.get("/v2/schema")
+async def get_schema_from_bucket_with_guid(
+    guid: str = None,
+    schema_bucket_repository: SchemaBucketRepository = Depends(),
+    schema_processor_service: SchemaProcessorService = Depends(),
+) -> dict:
+    """
+    Gets the filename of the bucket schema metadata and uses that to retrieve the schema metadata
+    with specific guid from the bucket
+
+    Parameters:
+    guid (str): GUID of the schema.
+    schema_firebase_repository (SchemaFirebaseRepository): injected dependency for
+        interacting with the schema collection in firestore.
+    schema_processor_service (SchemaProcessorService): injected dependency for
+        interacting with the schema collection in firestore.
+    """
+    logger.info("Getting bucket schema metadata...")
+    logger.debug(f"Input data: guid={guid}")
+
+    QueryParameterValidatorService.validate_guid_from_get_schema(guid)
+
+    bucket_schema_filename = (
+        schema_processor_service.get_schema_bucket_filename_from_guid(guid)
+    )
+
+    if not bucket_schema_filename:
+        logger.error("Schema metadata not found")
+        raise exceptions.ExceptionNoSchemaFound
+
+    logger.info("Bucket schema location successfully retrieved.")
+    logger.debug(f"Bucket schema location: {bucket_schema_filename}")
```

Comment:
Same here with condensing the L122 and L124 Info logs into one info log, the L123 debug should still remain as is.